zokrates_cli/src/bin.rs

```
@@ -691,76 +740,111 @@ fn cli() -> Result<(), String> {
             let curve = sub_matches.value_of("curve").unwrap();
+            let backend = sub_matches.value_of("backend").unwrap();
             let proof_system = sub_matches.value_of("proving-scheme").unwrap();
 
-            match proof_system {
-                constants::G16 => match curve {
-                    constants::BN128 => cli_export_verifier::<Bn128Field, G16>(sub_matches)?,
-                    constants::BLS12_381 => cli_export_verifier::<Bls12Field, G16>(sub_matches)?,
-                    _ => unimplemented!(),
-                },
-                #[cfg(feature = "libsnark")]
-                constants::PGHR13 => match curve {
-                    constants::BN128 => cli_export_verifier::<Bn128Field, PGHR13>(sub_matches)?,
-                    _ => unimplemented!(),
-                },
+            match backend {
+                constants::BELLMAN => match proof_system {
+                    constants::G16 => match curve {
+                        constants::BN128 => cli_export_verifier::<Bn128Field, G16>(sub_matches),
+                        constants::BLS12_381 => cli_export_verifier::<Bls12Field, G16>(sub_matches),
+                        _ => unimplemented!(),
+                    },
+                    _ => Err(format!(
+                        "Proving scheme {} is not supported by {} backend",
+                        proof_system, backend
+                    )),
+                }?,
                 #[cfg(feature = "libsnark")]
-                constants::GM17 => match curve {
-                    constants::BN128 => cli_export_verifier::<Bn128Field, GM17>(sub_matches)?,
-                    _ => unimplemented!(),
-                },
+                constants::LIBSNARK => match proof_system {
+                    constants::PGHR13 => match curve {
+                        constants::BN128 => cli_export_verifier::<Bn128Field, PGHR13>(sub_matches),
+                        _ => unimplemented!(),
```

this should return a clean error